com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/Denoising/DiffuseDenoiser.compute

```diff
@@ -165,30 +177,62 @@ void BILATERAL_FILTER(uint3 dispatchThreadId : SV_DispatchThreadID, uint2 groupT
     #endif
 }
 
-// TODO: This needs to check depth before upscale to avoid artifacts
+#define GATHER_REGION_SIZE DIFFUSE_DENOISER_TILE_SIZE
+#define GATHER_REGION_SIZE_2 (GATHER_REGION_SIZE * GATHER_REGION_SIZE)
+groupshared uint gs_cacheLighting[GATHER_REGION_SIZE_2];
+groupshared float gs_cacheLuminance[GATHER_REGION_SIZE_2];
+groupshared float gs_cacheDepth[GATHER_REGION_SIZE_2];
+
+void FillGatherDataLDS(uint groupIndex, uint2 pixelCoord)
+{
+    int2 sampleCoord = int2(clamp(pixelCoord.x, 0, _ScreenSize.x - 1), clamp(pixelCoord.y, 0, _ScreenSize.y - 1));
+    #ifdef SINGLE_CHANNEL
+    gs_cacheLuminance[groupIndex] = LOAD_TEXTURE2D_X(_DenoiseInputTexture, sampleCoord).x;
+    #else
+    float3 lighting = LOAD_TEXTURE2D_X(_DenoiseInputTexture, sampleCoord).xyz;
+    gs_cacheLighting[groupIndex] = PackToR11G11B10f(lighting);
+    #endif
+    float depthValue = LOAD_TEXTURE2D_X(_DepthTexture, sampleCoord).x;
+    gs_cacheDepth[groupIndex] = depthValue;
+}
+
+uint OffsetToLDSAdress(uint2 groupThreadId, int2 offset)
+{
+    // Compute the tap coordinate in the 8x8 grid
+    uint2 tapAddress = (uint2)((int2)(groupThreadId) + offset);
+    return clamp((uint)(tapAddress.x) + tapAddress.y * GATHER_REGION_SIZE, 0, GATHER_REGION_SIZE_2 - 1);
+}
+
 [numthreads(DIFFUSE_DENOISER_TILE_SIZE, DIFFUSE_DENOISER_TILE_SIZE, 1)]
-void GATHER_FILTER(uint3 dispatchThreadId : SV_DispatchThreadID, uint2 groupThreadId : SV_GroupThreadID, uint2 groupId : SV_GroupID)
+void GATHER_FILTER(uint3 centerCoord : SV_DispatchThreadID, int groupIndex : SV_GroupIndex, uint2 groupThreadId : SV_GroupThreadID, uint2 groupId : SV_GroupID)
 {
-    UNITY_XR_ASSIGN_VIEW_INDEX(dispatchThreadId.z);
+    UNITY_XR_ASSIGN_VIEW_INDEX(centerCoord.z);
 
-    // Fetch the current pixel coordinate
-    uint2 centerCoord = groupId * DIFFUSE_DENOISER_TILE_SIZE + groupThreadId;
+    // Fill color and lighting to the LDS
+    FillGatherDataLDS(groupIndex, centerCoord.xy);
+
+    // Make sure all values are loaded in LDS by now.
+    GroupMemoryBarrierWithGroupSync();
+
+    // Read the high res depth
+    int outputIdx = OffsetToLDSAdress(groupThreadId, int2(0, 0));
+    float targetDepth = gs_cacheDepth[outputIdx];
+
+    // Compute the 2x2 pixelregioncorner
+    uint2 corner = centerCoord.xy - uint2(centerCoord.x & 1, centerCoord.y & 1);
+    uint2 cornerGroupThread = corner - groupId * DIFFUSE_DENOISER_TILE_SIZE;
+
+    // Grab the indices of the sub-region to use
+    int ldsIdx0 = OffsetToLDSAdress(cornerGroupThread, int2(0, 0));
```

Comment:
So I see you don't load a safety border so it is clamp at _group_ border which I guess can give problems if we have signficiant depth discontinuities happening at the edge of the group? Might be ok, but leaving comment here to see if needs further discussion or not.

Reply:
Not sure to get what you are describing, let's chat